When install @slidev/cli globally, we are forced to globally install @slidev/client and katex

<!-- ⚠️ Please DON'T ignore the issue template -->

<!-- 💡 Consider upgrading to the latest version before sending the issue -->

**Describe the bug**
https://github.com/slidevjs/slidev/blob/2acab57e93b069a35541a4598df37a6c576b1d28/packages/slidev/node/plugins/extendConfig.ts#L57-L60
The L58 code resolves to extenal dependancy of slidev/cli and not internal ones, running slidev trigger the following error :
```
Error: Cannot find module '/home/alex/.local/lib/node_modules/@slidev/client/package.json'
```
But client is already install as internal dependancy in :
```
/home/alex/.local/lib/node_modules/@slidev/cli/node_modules/@slidev/client/
```

This force us to install @slidev/client and katex globally to get rid of the module error (but i think is not the intended behavior).

**To Reproduce**
Steps to reproduce the behavior:
1. npm install -g @slidev/cli
2. create slides.md
3. slidev
4. answer yes to install default theme globally

**Desktop (please complete the following information):**
 - OS: arch linux
 - Browser: firefox
 - Slidev version: 0.27.13

--- a/packages/slidev/node/plugins/extendConfig.ts
+++ b/packages/slidev/node/plugins/extendConfig.ts
@@ -1,12 +1,11 @@
 import { dirname, join } from 'path'
 import { InlineConfig, mergeConfig, Plugin } from 'vite'
 import isInstalledGlobally from 'is-installed-globally'
-import resolveGlobal from 'resolve-global'
 import { uniq } from '@antfu/utils'
 import { getIndexHtml } from '../common'
 import { dependencies } from '../../../client/package.json'
 import { ResolvedSlidevOptions } from '../options'
-import { resolveImportPath, toAtFS } from '../utils'
+import { resolveImportPath, resolveGlobalImportPath, toAtFS } from '../utils'
 import { searchForWorkspaceRoot } from '../vite/searchRoot'
 
 const EXCLUDE = [
@@ -55,7 +54,7 @@ export function createConfigPlugin(options: ResolvedSlidevOptions): Plugin {
               searchForWorkspaceRoot(options.cliRoot),
               ...(
                 isInstalledGlobally
-                  ? [dirname(resolveGlobal('@slidev/client/package.json')), dirname(resolveGlobal('katex/package.json'))]
+                  ? [dirname(resolveGlobalImportPath('@slidev/client/package.json')), dirname(resolveGlobalImportPath('katex/package.json'))]
                   : []
               ),
             ]),

--- a/packages/slidev/node/utils.ts
+++ b/packages/slidev/node/utils.ts
@@ -1,7 +1,8 @@
+import { join } from 'path'
 import { ensurePrefix, slash } from '@antfu/utils'
 import isInstalledGlobally from 'is-installed-globally'
 import { sync as resolve } from 'resolve'
-import resolveGlobal from 'resolve-global'
+import globalDirs from 'global-dirs'
 import type Token from 'markdown-it/lib/token'
 import { ResolvedFontOptions } from '@slidev/types'
 
@@ -21,7 +22,12 @@ export function resolveImportPath(importName: string, ensure = false) {
 
   if (isInstalledGlobally) {
     try {
-      return resolveGlobal(importName)
+      return require.resolve(join(globalDirs.yarn.packages, importName))
+    }
+    catch {}
+
+    try {
+      return require.resolve(join(globalDirs.npm.packages, importName))
     }
     catch {}
   }
@@ -30,6 +36,25 @@ export function resolveImportPath(importName: string, ensure = false) {
     throw new Error(`Failed to resolve package "${importName}"`)
 
   return undefined
+}
+
+export function resolveGlobalImportPath(importName: string): string {
+  try {
+    return resolve(importName, { preserveSymlinks: false, basedir: __dirname })
+  }
+  catch {}
+
+  try {
+    return require.resolve(join(globalDirs.yarn.packages, importName))
+  }
+  catch {}
+
+  try {
+    return require.resolve(join(globalDirs.npm.packages, importName))
+  }
+  catch {}
+
+  throw new Error(`Failed to resolve global package "${importName}"`)
 }
 
 export function stringifyMarkdownTokens(tokens: Token[]) {

--- a/packages/slidev/package.json
+++ b/packages/slidev/package.json
@@ -59,6 +59,7 @@
     "debug": "^4.3.2",
     "fast-glob": "^3.2.7",
     "fs-extra": "^10.0.0",
+    "global-dirs": "^3.0.0",
     "import-from": "^4.0.0",
     "is-installed-globally": "^0.4.0",
     "jiti": "^1.12.9",

